hax/hax/handler.py

```diff
@@ -110,7 +110,8 @@ def pull_msg():
                                      status=ServiceHealth.FAILED)])
                         LOG.debug('waiting for broadcast of %s for ep: %s',
                                   ids, item.remote_rpc_endpoint)
-                        self.herald.wait_for_all(HaLinkMessagePromise(ids))
+                        self.herald.wait_for_delivery(
```

Comment:
I would still prefer to have 2 separate explicitly named functions, wait_for_any() and wait_for_all(). You can rather take out the common content of the functions into a separate local function which can be invoked from both of them.

Reply:
Done.